lib/freeze.js

```diff
@@ -37,42 +34,53 @@ module.exports = class Freeze {
     });
   }
 
-  freeze(context, props) {
+  async freeze(context, props) {
     const labels = context.payload.issue.labels;
     if (GitHubHelper.isLabelOnIssue(
       context.payload.issue,
       this.config.labelName) === undefined) {
       labels.push(this.config.labelName);
     }
 
-    this.github.issues.edit(context.issue({
+    const kv = metadata(context.github, context.issue(), process.env.APP_ID);
+    await kv.set('snooze', props);
+
+    await this.github.issues.edit(context.issue({
       state: 'closed',
       labels
     }));
 
-    this.github.issues.createComment(context.issue({
+    await this.github.issues.createComment(context.issue({
       body: 'Sure thing. I\'ll close this issue for a bit. I\'ll ping you around ' +
-      moment(props.unfreezeMoment).calendar() + ' :clock1: ' +
-      '<!-- ' + JSON.stringify(props) + '-->'
+      moment(props.unfreezeMoment).calendar() + ' :clock1: '
     }));
   }
 
-  unfreeze(issue, props) {
+  async checkUnfreeze(issue) {
+    const kv = metadata(this.github, issue, process.env.APP_ID);
```

Comment:
should metadata take context as a param so that it gets both `.github` and the `app_id` in one-fell swoop? (I assume context knows of app_id because it created `.github`...)

Reply:
Good point. I really like the the API being something like:

```js
metadata(context).get(key)
```

`context` doesn't actually know about `APP_ID` since it is handed the github client, but…every payload includes an `installation.id`, which can be used for namespacing.